src/utils/retry_utils/src/retry_utils.c

```diff
@@ -186,14 +186,15 @@ int RetryUtils_GetRetryParamsMapSize()
 }
 
 void ReadRetryParamsArrayFromAgentConfigJson(
-    ADUC_Retriable_Operation_Context* context, JSON_Value* agentJsonValue, int retryParamsMapSize)
+    ADUC_Retriable_Operation_Context* context, JSON_Value* agentRetrySettingsJsonValue, int retryParamsMapSize)
 {
     const char* infoFormatString = "Failed to read '%s.%s' from agent config. Using default value (%d)";
 
     for (int i = 0; i < retryParamsMapSize; i++)
     {
         ADUC_Retry_Params* params = &context->retryParams[i];
-        JSON_Value* retryParams = json_object_dotget_value(json_object(agentJsonValue), s_retryParamsMap[i].name);
+        JSON_Value* retryParams =
```

Comment:
immediately casting agentRetrySettingsJsonValue to json_object without checking inputs nor the object it's casted to.